src/client/directory.ts

```diff
@@ -10,60 +10,150 @@
+ *
+ * This type alias exists to disambiguate raw string app identity from the [[AppId]].
+ */
 export type AppName = string;
 
 /**
- * An application in the app directory
+ * An application in the app directory.
  */
 export interface Application {
+    /**
+     * The Application Identifier. Please see https://fdc3.finos.org/docs/1.0/appd-discovery#application-identifier.
+     */
     appId: AppId;
+    /**
+     * The human readable app name.
+     */
     name: AppName;
+    /**
+     * An application manifest, used to launch the app. This should be a URL that points to an OpenFin json manifest.
+     */
     manifest: string;
+    /**
+     * The manifest type. Always 'openfin'.
+     */
     manifestType: string;
-
+    /**
+     * The version of the app. Please use [semantic versioning](https://en.wikipedia.org/wiki/Software_versioning).
+     */
     version?: string;
+    /**
+     * The human-readable title of the app, typically used by the launcher UI. If not provided, the name is used.
+     */
     title?: string;
+    /**
+     * Tooltip used by any UIs that display app information. A short explanatory text string.
+     */
     tooltip?: string;
+    /**
+     * Longer description of the app.
+     */
     description?: string;
+    /**
+     * Images that can be displayed as part of the app directory entry. Use these for screenshots, previews or similar. These are not the
+     * application icons: use 'icons' for that.
+     */
     images?: AppImage[];
+    /**
+     * Contact email address.
+     */
     contactEmail?: string;
+    /**
+     * Support email address.
+     */
     supportEmail?: string;
+    /**
+     * Name of the publishing company, organization, or individual.
+     */
     publisher?: string;
-
-    signature?: string;
+    /**
+     * Icons used in the app directory display. A launcher may be able to use various sizes.
+     */
     icons?: Icon[];
+    /**
+     * Additional config. Currently unused by the OpenFin implementation.
+     */
     customConfig?: NameValuePair[];
-
-    intents?: Intent[];
+    /**
+     * The set of intents associated with this AppDir entry.
+     */
+    intents?: AppDirIntent[];
 }
-
-interface AppImage {
+/**
+ * An image for an app in the app directory.
+ */
+export interface AppImage {
+    /**
+     * A URL that points to an image.
+     */
     url: string;
+    /**
+     * Alt text to be displayed with the image.
+     */
     tooltip?: string;
+    /**
+     * Additional text description.
+     */
     description?: string;
 }
-interface Icon {
+
+/**
+ * An icon for an app in the app directory.
+ */
+export interface Icon {
+    /**
+     * A URL that points to an icon.
+     */
     icon: string;
 }
 
-interface NameValuePair {
+/**
+ * The application allows extra configuration to be passed in, and uses an array of NameValuePairs,
+ * where the key is the name.
+ */
+export interface NameValuePair {
+    /**
+     * Key for the extra configuration.
```

Comment:
I think we'd be fine without these property comments since they're described in the interface comment anyway, but should probably discuss as a group whether we need a comment on every single thing.

Reply:
I'd actually rather lose these too. I put them to in to catch 'em all. I'll delete them